frontend/openchat-agent/src/services/user/user.client.interface.ts

```diff
@@ -67,15 +68,15 @@ export interface IUserClient {
     sendMessage(
         recipientId: string,
         sender: CreatedUser,
-        message: Message,
+        event: EventWrapper<Message>,
         replyingToChatId?: string,
         threadRootMessageIndex?: number
     ): Promise<[SendMessageResponse, Message]>;
     sendGroupICPTransfer(
         groupId: string,
         recipientId: string,
         sender: CreatedUser,
-        message: Message,
+        event: EventWrapper<Message>,
```

Comment:
Seems a little iffy given it is only needed for the caching layer

Reply:
Yes it is a bit iffy - and in fact I'm not convinced I actually _do_ need the whole event now. Might be able to do a bit of a re-jig there.

Reply:
Come to think of it- there's no easy way round it. We _do_ need to capture the whole event rather than just the message. Otherwise we won't be able to splice them into the timeline.